### ISSUE TITLE
"Error loading KCT Data" error message.

### ISSUE THREAD

**Issue by Rodhern** [magico13/KCT#78]
First of, thank you for a great mod. Ran into an error message today (and it said to report it as a bug)

Here is my save game
 http://www.act-consult.net/Rodhern/Games/KSP/Temp1/persistent.sfs
and here is the log file
 http://www.act-consult.net/Rodhern/Games/KSP/Temp1/KSP.log

I load the game, click the runway, select a vessel and a pilot, click launch.
In the 'simulation' scene I click "build it" and "close" from the KCT window.
Now I click the (stock) "Space Center" big blue button at the top.
After a second or so I get a "Error loading KCT Data" message telling me that the game is now broken.

B.t.w. I use a custom BPFormula in KCT_Formulas.cfg, but the error is the same regardless.

**Comment by magico13** [comment]
Well, you've certainly found an error within KCT, which isn't frequently the case with the OnLoad error message (usually another mod has an error, which then prevents other mods from reading their data). The issue is with how you're running the simulation. By not using the KCT methods to set it up in the editor KCT doesn't set the "launchedVessel" variable, so when it tries to make a copy of that to build it, it can't (because it doesn't exist).

For now, don't try to build vessels that way. I'm not sure exactly how I'll go about fixing this. I might be able to set the launchedVessel from the craft file, or it might have to wait for me to create my own GUI that pops up when clicking the launchpad/runway.

**Comment by Rodhern** [comment]
Sure, I did figure that I probably shouldn't use the launchpad/runway to order builds.

It is almost embarrassing how long it took me to figure out the correct way to recover a vessel to storage (you must first click fly). With this in mind maybe the important lesson is to tell with more emphasis what the correct buttons for building and recovering are (I had override launch button enabled, but will disable that now).

I would prefer that you don't make a new GUI for the launchpad/runway, because I find the existing one nice to use as a vehicle library, scrolling through the designs and reading the descriptions.

Basically the only fix that is needed is that the "Build It!" in this situation should do nothing, apart from maybe tell in a message that it does nothing :-) .

**Comment by Rodhern** [comment]
Ohh, just realized that it is actually a cheat, in that I get the simulation for free when entering via the launchpad/runway. Oh, well, done is done. :-)

**Comment by magico13** [comment]
The purpose of the new GUI would be to let you order builds and perform simulations on saved craft. You'd still see their descriptions and info (like cost and size) and would be able to assign Kerbals, but it might not be as pretty as the Stock one. Basically, it would serve the same purpose as the Stock one, but would actually work with KCT instead of against it ;) As you've discovered, the simulations that start through the runway/launchpad don't work properly at ALL and I've only been able to make them start a "simulation", otherwise you'd be able to launch things without spending time. That's why I need to somehow override things (either my own GUI or to hack the Stock one)

You should keep "Override Launch Button" on, as that's talking about the launch button in the editor (which is the only place you should order builds or simulate from, except when you Duplicate a craft to order a new copy, which can be done in the Space Center).

Recovering to storage is still non-standard. Normal recovery will always break a ship into parts and put them in the Inventory. Recovering to storage is a buggy mess as KSP really doesn't like going from a Vessel to a Ship Construct. Quite often it can break your craft, or even break your save.

The "correct" buttons to build/simulate are the Launch button in the editor. The "correct" button to recover a ship safely is the stock recover buttons. The "correct' button to recover a ship in a potentially game breaking way (that everyone really really wanted :P ) is the "Recover to Storage" button.

**Comment by magico13** [comment]
Modding KSP is weird, you basically have to break it to do a lot of the things I try to do and they keep adding new features that potentially make things harder :P

**Comment by Rodhern** [comment]
> You should keep "Override Launch Button" on, as that's talking about the launch button in the editor ...
I notice that when "Override Launch Button" is off then the stock launch button is disabled from the editor; that is kind of a cool way to tell one to look elsewhere for the proper button. If the same could be done from the launchpad/runway pop-up the problem would be solved.

**Comment by magico13** [comment]
> that is kind of a cool way to tell one to look elsewhere for the proper button

Ironically, most people hate the separate button and want more things to be integrated into the stock buttons. I might be able to override the function for the runway/launchpad to pop up a message saying to build from the editor and launch from the KCT window, though.

**Comment by DiamondExcavater** [comment]
http://forum.kerbalspaceprogram.com/threads/92377-1-0-4-Kerbal-Construction-Time-1-2-2-%288-26-15%29-Unrapid-Planned-Assembly?p=2229493&viewfull=1#post2229493
I have the same issue, except it's on start when loading save
